src/main/java/org/jenkinsci/plugins/codesonar/services/CodeSonarCacheService.java

```diff
@@ -30,14 +32,26 @@ public class CodeSonarCacheService {
     private MetricsService metricsService;
     private ProceduresService proceduresService;
     private AlertsService alertsService;
+    private WarningsService warningsService;
     private Map<Pair<URI,Long>, CodeSonarAnalysisData> cachedAnalyses;
 
-    public CodeSonarCacheService(HttpService httpService, CodeSonarHubInfo hubInfo) {
+    private CodeSonarCacheService(HttpService httpService, CodeSonarHubInfo hubInfo) {
         this.httpService = httpService;
         this.hubInfo = hubInfo;
         cachedAnalyses = new HashMap<>();
     }
     
+    public static CodeSonarCacheService createInstance(HttpService httpService, CodeSonarHubInfo hubInfo) {
+        if(instance == null) {
+            instance = new CodeSonarCacheService(httpService, hubInfo);
+        }
+        return instance;
+    }
+    
+    public static CodeSonarCacheService getInstance() {

```

Comment:
This method is specifically for those that need to get the instance, whenever already present, but which don't have the ability to instantiate a new one if not present.